### PR TITLE
[RFR] Create a multiplayer game

### PR DESCRIPTION
**Pull request by Luwangel**
- [x] Add the create button
- [x] Create the waiting option

### CODE REVIEW

File: __tests__/src/components/button.spec.js [hunk 1]
```diff
@@ -6,16 +6,20 @@ import Button from '../../../src/components/button';
...
-        const component = renderer.create(<Button icon="" label="" route="" />);
+        const component = renderer.create(
+            <Button icon="" label="" route="" onClick={() => {}} />,
+        );
         const tree = component.toJSON();
         expect(tree).toMatchSnapshot();
     });
 
-    test('should contain a link', () => {
+    test('should contain a link with', () => {
```
Comment (floo51) [inline __tests__/src/components/button.spec.js]: `with` what ?

File: pages/game.js [hunk 2]
```diff
@@ -18,6 +18,16 @@ import {
 } from '../src/core/game';
 import { shuffle } from '../src/core/shuffler';
 
+export const title = (isLoading, isVictory, turn) => {
```
Comment (floo51) [inline pages/game.js:21]: Is there any reason you export it ? It doesn't seem tested or used anywhere.

File: pages/multiplayerGame.js [hunk 3]
```diff
@@ -11,11 +11,37 @@ import Page from '../src/components/page';
 import Row from '../src/components/row';
 import Section from '../src/components/section';
 
-import { game, move } from '../src/services/multiplayerGameService';
+import { game, join, move } from '../src/services/multiplayerGameService';
+
+export const title = (
```
Comment (floo51) [inline pages/multiplayerGame.js]: Same here

File: pages/multiplayerGames.js [hunk 4]
```diff
@@ -3,13 +3,14 @@ import React, { Component } from 'react';
...
-import ListGames from '../src/components/listGames';
 import Button from '../src/components/button';
+import ListGames from '../src/components/listGames';
+import Image from '../src/components/image';
 import Page from '../src/components/page';
 import Row from '../src/components/row';
 import Section from '../src/components/section';
 
-import { games, join } from '../src/services/multiplayerGameService';
+import { games, join, newGame } from '../src/services/multiplayerGameService';
```
Comment (floo51) [inline pages/multiplayerGames.js]: `newGame` is badly named, it's a factory taking some config (API url) to build a method actually starting a new game.

**Review by floo51 — CHANGES_REQUESTED** [review]

File: pages/multiplayerGame.js [hunk 5]
```diff
@@ -27,13 +53,15 @@ export default class MultiplayerGame extends Component {
...
     };
 
-    static getInitialProps = async ({ query }) => ({
-        id: query.id,
-        token: query.token,
-    });
+    static getInitialProps = async ({ query }) => {
+        return {
+            id: query.id,
+            token: query.token || '',
```
Comment (jdemangeon) [inline pages/multiplayerGame.js]: It's better to use null as default empty value instead of an empty string.

File: pages/multiplayerGame.js [hunk 6]
```diff
@@ -98,40 +128,58 @@ export default class MultiplayerGame extends Component {
...
     componentWillMount() {
-        Router.pushRoute('multiplayer_game', {}, { shallow: true });
-
         const { id, token } = this.props;
+
+        if (id === -1) {
+            Router.pushRoute('multiplayer_games');
+            return;
+        }
+        if (token === '') {
```
Comment (jdemangeon) [inline pages/multiplayerGame.js]: !token (with null)

File: src/components/button.js [hunk 7]
```diff
@@ -4,27 +4,33 @@ import { Link } from '../routes';
...
 
-const Button = ({ color, icon, label, route }) => {
+const Button = ({ color, icon, label, onClick, route }) => {
     const buttonClass = ClassNames({
         btn: true,
         [color]: color,
     });
 
-    return (
+    return route ? (
```
Comment (jdemangeon) [inline src/components/button.js]: No need for ternary here. 

You should do if() { return ... } ..... return  ... instead

File: pages/multiplayerGames.js (hunk 4 above)
Comment (Luwangel) [reply pages/multiplayerGames.js]: So it's the case for all methods in this service.

Should I use gamesFactory, joinFactory, newGameFactory ?

Comment (floo51) [reply pages/multiplayerGames.js]: Mostly depends on the naming rules of your project.  But even if the word `factory` usually scares developpers, it describes perfectly what it is there.

File: pages/game.js (hunk 2 above)
Comment (Luwangel) [reply pages/game.js:21]: I've added more tests

**Review by floo51 — APPROVED** [review]

**Pull request merged**